app/src/pages/OnDeviceDisplay/ProtocolDashboard/ProtocolRow.tsx

```diff
@@ -0,0 +1,210 @@
+import * as React from 'react'
+import { useDispatch, useSelector } from 'react-redux'
+import { useHistory } from 'react-router-dom'
+import { useTranslation } from 'react-i18next'
+import styled from 'styled-components'
+import { format, formatDistance } from 'date-fns'
+import {
+  ALIGN_CENTER,
+  COLORS,
+  DIRECTION_COLUMN,
+  Flex,
+  Icon,
+  JUSTIFY_CENTER,
+  SPACING,
+  TEXT_ALIGN_CENTER,
+  TYPOGRAPHY,
+  truncateString,
+  useLongPress,
+} from '@opentrons/components'
+
+import { StyledText } from '../../../atoms/text'
+import { ModalShell } from '../../../molecules/Modal'
+import { getPinnedProtocolIds, updateConfigValue } from '../../../redux/config'
+
+import type { Dispatch } from '../../../redux/types'
+import type { PinnedProtocolIds } from '../../../redux/config/types'
+
+import {
+  useCreateRunMutation,
+  // TODO useDeleteProtocolMutation,
+} from '@opentrons/react-api-client'
+import type { ProtocolResource } from '@opentrons/shared-data'
+
+const TableRow = styled('tr')`
+  border: 1px ${COLORS.medGreyHover} solid;
+  height: 4rem;
+  padding: ${SPACING.spacing5};
+`
+
+const TableDatum = styled('td')`
+  padding: ${SPACING.spacing2};
+  white-space: break-spaces;
+  text-overflow: wrap;
+`
+
+export function ProtocolRow(props: {
+  protocol: ProtocolResource
+  lastRun: string | undefined
```

Comment:
probably mostly we use ? instead of `| undefined`
```suggestion
  lastRun?: string
```